server.go

```diff
@@ -1500,6 +1506,13 @@ func (s *Server) ReadSeries(database, retentionPolicy, name string, tags map[str
 // Returns a resultset for each statement in the query.
 // Stops on first execution error that occurs.
 func (s *Server) ExecuteQuery(q *influxql.Query, database string, user *User) Results {
+	// Authorize user to execute the query.
+	if err := Authorize(user, q, database); err != nil {
```

Comment:
By doing this here, if the user has the privilege to execute the first query, but not the second, none of the statements will execute. But only the second generated an error. Strictly speaking, this isn't what we agreed in the API -- the first statement should execute fine, but the second should not. Therefore shouldn't we push this "authorize" check in the loop below?

It may also have the nice side-effect of making `Authorize()` a little simple since it won't have to loop.